src/engine/sparqlExpressions/DateExpressions.cpp

```diff
@@ -53,31 +59,50 @@ inline auto extractStrTimezone =
   return Id::makeUndefined();
 };
 
+//______________________________________________________________________________
+inline auto extractTimezoneDurationFormat =
+    [](std::optional<DateYearOrDuration> d) {
+      if (d.has_value()) {
```

Comment:
Start with
`if !d.has_value() {return undefined';}`
Then the rest of the code is not nested inside an If.

And add a `TODO<C++23> Use the monadic operation for `std::optional`.